device/device.go

```diff
@@ -0,0 +1,110 @@
+package device
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/device/deviceType"
+	"github.com/josephroberts/edge-node-manager/radio/bluetooth"
+	"github.com/josephroberts/edge-node-manager/radio/radioType"
+	"github.com/josephroberts/edge-node-manager/radio/wifi"
+	"github.com/josephroberts/edge-node-manager/radio/zigbee"
+)
+
+type Type struct {
+	Device deviceType.Type
+	Radio  radioType.Type
+}
+
+func (t Type) Scan(name string, timeout time.Duration) ([]string, error) {
+	switch t.Radio {
+	case radioType.BLUETOOTH:
+		return bluetooth.Scan(name, timeout)
+	case radioType.WIFI:
+		return wifi.Scan(name, timeout)
+	case radioType.ZIGBEE:
+		return zigbee.Scan(name, timeout)
+	default:
+		return nil, fmt.Errorf("Radio %s does not exist", t.Radio)
+	}
+}
+
+func Create(t *Type) Interface {
```

Comment:
If you're going to be using types to have specific methods for specific devices then, no unless you use reflection which I think would be overengineered.